### PR TITLE
Added summary statistics in output tables for date variables

### LINKED ISSUE #7910: Is the latest version of One Variable > Summarise > Customised installed?
@lilyclements this seems to work well in that it never crashes.  But I thought it could now produce some summaries for (ordered) factors or dates?
With the dodoma data the dates give the following results:
image

I thought you had fixed it, so that minimum, maximum, etc were given, and even shown as dates?

With the ordered factors in the diamonds data I had wondered whether the minimum and maximum might be given - and the n-distinct would give the number of levels?

image

### PR DESCRIPTION
**Pull request by lilyclements**
Adds in changes from #7465 that were lost in replacement. Fixes #7910 

@africanmathsinitiative/developers this is ready to review

### CODE REVIEW

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@rdstern if you can test/approve, then we can merge, thanks

**Comment by rdstern** [comment]
@lilyclements I added comments into #7910.  Here they are:
I still don't see a difference in the One Variable summarise - maybe they haven't been added in there yet. 
 For ordered factors with one variable I still get:
image

or for dates:
image

So I wondered if perhaps the changes were added to the general dialogue.  So I tried this:

image
And it gave me this error:

image
The commands that gave the error were:
```
summary_table <- data_book$summary_table(data_name="dodoma", columns_to_summarise="date", treat_columns_as_factor=FALSE, j=1, summaries=c("summary_mean", "summary_min", "summary_max", "p25", "p75", "summary_median"))


last_table <- (mmtable2::mmtable(data=summary_table, cells=value) + mmtable2::header_top_left(variable='summary-variable'))
data_book$add_table(table_name="last_table", table=last_table, data_name="dodoma")
data_book$get_tables(data_name="dodoma", table_name="last_table")

rm(list=c("last_table", "summary_table"))

```

**Review by rdstern — DISMISSED** [review]
@lloyddewit this seems to work fine, so I am approving.
I found a different problem that the option in the dialogue to order the summaries seems to be ignored in the output.  But that is for another issue

File: instat/static/InstatObject/R/Backend_Components/summary_functions.R [hunk 1]
```diff
@@ -178,8 +176,9 @@ DataBook$set("public", "calculate_summary", function(data_name, columns_to_summa
       #TODO result_names could be horizontal/vertical vector, matrix or single value
       else result_name <- result_names[i,j]
       if(percentage_type == "none") {
+          summary_function_exp <- paste0(summary_type, "(x = ", column_names, function_exp)
```
Comment (lloyddewit) [inline instat/static/InstatObject/R/Backend_Components/summary_functions.R:179]:
```suggestion
        summary_function_exp <- paste0(summary_type, "(x = ", column_names, function_exp)
```

**Review by lloyddewit — CHANGES_REQUESTED** [review]

**Review by lloyddewit — APPROVED** [review]

**Review by rdstern — COMMENTED** [review]
@lilyclements and @lloyddewit I wonder what has happened.  I thought it worked before.  Now when I try with the Dodoma data.  I first got:

image

Oddly it did give me the sd!  When I asked for more I got an error:
Here is the dialogue:
image
And it gave me:

image

I wondered if that was caused, because there was no factor - but that had worked before.  I got the same error when I added the factor.

I tried the script and noticed that the key line was repeated twice, though I think this might now be caused by other changes?

```
# Code généré par la boîte de dialogue, Frequency/Summary Tables

summary_table <- data_book$summary_table(data_name="dodoma", columns_to_summarise="date", treat_columns_as_factor=FALSE, j=1, summaries=c("summary_mean", "summary_max", "summary_min", "summary_count_non_missing", "p25", "p75", "summary_median"))

summary_table <- data_book$summary_table(data_name="dodoma", columns_to_summarise="date", treat_columns_as_factor=FALSE, j=1, summaries=c("summary_mean", "summary_max", "summary_min", "summary_count_non_missing", "p25", "p75", "summary_median"))
last_table <- (mmtable2::mmtable(data=summary_table, cells=value) + mmtable2::header_left_top(variable='summary-variable') + mmtable2::header_top_left(variable=month_abbr))
data_book$add_object(data_name="dodoma", object_name="last_table", object_type_label="table", object_format="html", object=last_table)
data_book$get_object(data_name="dodoma", object_name="last_table")
rm(list=c("last_table", "summary_table"))

```

**Comment by lilyclements** [comment]
@rdstern the issue is because of the missing values. If you check "Omit missing values" does this still occur?

@rdstern looking at the code you posted, we have `month_abbr` in the `mmtable2` part of the code but not in the `data_book$summary_table` bit of the code. I think therefore the issue is in the visual studio code not reading in the arguments in the factors receiver?

**Comment by rdstern** [comment]
@lilyclements aaah I forgot about the missing possibility - I'll check that again.  Sorry.

**Review by rdstern — APPROVED** [review]

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
> @lilyclements Thank you for this PR. The PR title is automatically included in the release notes, so it should be meaningful for the user, rather than a technical description of the change. Can you think of a PR title that makes clear to the user which bug(s) have been fixed? Thanks

I changed the PR title to reflect what I think has improved for the user. If I misunderstood, then please let me know, thanks

**Comment by lilyclements** [comment]
> I changed the PR title to reflect what I think has improved for the user. If I misunderstood, then please let me know, thanks

Apologies I missed this comment. Thank you for the title improvement

**Pull request merged**